### ISSUE TITLE
What is the selection criteria of D in HSMM?

### ISSUE THREAD

**Issue by Wusir2018** [dylanebert/hsmm#1]
Hello,
Thank you for your open code!
What is the selection criteria for setting value of D? Why set D to maximum segment length in HSMM ?
If D is set as the maximum segment length, then the state corresponding to some smaller segment must also last for D time to jump to the next state? Compared with HMM, will the result be worse?
Could you help me?
Thank you very much!
Best wishes!

**Comment by dylanebert** [comment]
Hi, what do you mean by D? Are you referring to the maximum segment length (max_k in the code)?
If so, then it's just the maximum length, but the model can handle shorter segments, since it's a learned parameter (poisson_log_rates), the expected length for each state.

**Comment by Wusir2018** [comment]
@dylanebert 
Hello，Thank you for you reply!
I'm sorry I didn't describe the problem clearly!
I found some paper using $\lambda= [\pi, A, B, D]$ to describe HSMM model. And many researches use the ` maximum segment length` to represent the `D`. 
![20200929104527](https://user-images.githubusercontent.com/51500032/94507721-f6676380-0242-11eb-8a10-287fd706a876.png)
I have a question:  Many researchers describe the transition to the next state only when the duration becomes zero. If `D` is set as the maximum segment length, then the state corresponding to some smaller segment must also last for `D` time to jump to the next state? Compared with HMM, will the result be worse?
$\pi$ : Initial state probability vector
$A$: State transition probability matrix
$B$: Observation distribution
$D$: State duration distribution

Question 2: Thank you for your published code! I think your code is very interesting. It combines the neural network with HSMM, but you haven't published your paper. May I ask how you combined the two? Could you published your paper?

Could you help me?
Thank you very much!
Best wishes!

reference: [https://www.cs.ubc.ca/~murphyk/Papers/segment.pdf](url)

**Comment by dylanebert** [comment]
Oh, your reference isn't quite how this implementation works. Duration here is a probability distribution, not a decrementing value. This code is very in-development still, and I haven't touched the neural network part or published anything yet, but I've based this code on the following paper: https://arxiv.org/abs/2005.03684
I recommend checking out that paper, which also has its code on github.

**Comment by Wusir2018** [comment]
@dylanebert 
Thank you very much!
Duration here is a probability distribution, it means that the current state has a probability of transitioning to the next state, rather than having to continuously transition to its own state？

We often use Poisson distribution or Gamma distribution to describe this duration, but how can we chose the parameters (k and \lambda) of Poisson distribution? How can we chose the size of duration matrix?
Possion distribution: $P(x=k)=\frac{\lambda^k}{k!} e^{-\lambda}$

Best wishes!

**Comment by dylanebert** [comment]
Yes, using a poisson distribution. The poisson parameters are learned - poisson_log_rates in hsmm.py
The discrete length probabilities are computed in length_log_probs().

**Comment by Wusir2018** [comment]
@dylanebert 
Thank you very much! :)
